feat: Add ability to view version of the individual OSCAL object

enikonovad opened the pull request:
## Types of changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [ ] Hot fix (emergency fix and release)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation (change which affects the documentation site)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Release (`develop` -> `main`)

## Quality assurance (all should be covered).

- [x] My code follows the code style of this project.
- [ ] Documentation for my change is up to date?
- [x] My PR meets testing requirements.
- [x] All new and existing tests passed.
- [x] All commits are signed-off.

## Summary

Added ability to view the metadata version of the individual OSCAL objects.

`trestle version -n <object_name> -t <type>` will return metadata version of that object.
`trestle version` has the same logic as previously

## Key links:

- [Sonar coverage](https://sonarcloud.io/dashboard?id=compliance-trestle)

# Before you merge

- Ensure it is a 'squash commit' if not a release.
- Ensure CI is currently passing
- Check sonar. If you are working for a fork a maintainer will reach out, if required.

Closes #1292

fsuits commented:
The validate command also allows model name and model type, but it finds the model using `get_models_of_type` in `model_utils`.  See line 62 in `validator.py`  This would allow support for all (and future) model types, but it does require spelling out the full model type.  See the `-h` help command for `validate`.

We could use some kind of lookup based on short names like `ssp` but if so the validate command should be similarly changed.

So I would avoid using the term 'folder' and just refer to model name and model type - with the full type.

For test coverage you can just generate models on the fly, set the version or whatever as you like, and save them out - then query the version.  This could even be done in a loop over all model types, generate random strings for version et al., and not rely on existing test data.

If there is a reason you only want to support a subset of types then the above may not apply, but I would aim to keep things consistent with `validate`.

enikonovad commented:
@fsuits , good suggestions will address it.

enikonovad commented:
@fsuits I have addressed the comment and now the `trestle version` will support all models similarly to `validate`.
Also did a small refactoring (mostly renaming) of existing functionality in `model_utils`. This class will need further revision.

As for the tests, I think we can postpone the generation of the models on the fly until the #1282 is completed

fsuits commented on trestle/core/commands/version.py:
```diff
@@ -30,7 +33,44 @@ class VersionCmd(CommandBase):
     name = 'version'
 
+    def _init_arguments(self) -> None:
+        self.add_argument('-n', '--name', help='Name of the OSCAL object (folder name)', default=None, type=str)
+        self.add_argument(
+            '-t',
+            '--type',
+            help='Name of the OSCAL type (catalog, profile, component-definition, system-security-plan, etc)',
+            default=None,
+            type=str
+        )
```
I would say "Name of the OSCAL model" (and not mention folder name).  'Object' sounds lower level than model - and this only deals with what we refer to as OSCAL models.  None of the CLI commands mentions folder name, so this would stand out in that regard.  Models have a type and they are stored in trestle somehow by name - so just refer to the model name and type.

And on line 41 I would just say "Type of the model being queried: (catalog ...)"  This emphasizes the -t as type, and I don't think we need to say "name of the type".

enikonovad replied:
done

fsuits reviewed (approved):
LGTM with minor comment
